docs/docs/10.9-perf.md

```diff
@@ -6,7 +6,7 @@ prev: pure-render-mixin.html
 next: advanced-performance.html
 ---
 
-React is usually quite fast out of the box. However, in situations where you need to squeeze every ounce of performance out of your app, it provides a [shouldComponentUpdate](/react/docs/component-specs.html#updating-shouldcomponentupdate) hook where you can add optimization hints to React's diff algorithm.
+React is usually quite fast out of the box. However, in situations where you need to squeeze every ounce of performance out of your app, it provides a [shouldComponentUpdate](/react/docs/component-specs.html#updating-shouldcomponentupdate) hook where you can add optimization hints to React's diff algorithms.
```

Comment:
Let's revert this. Sure it's probably technically a collection of algorithms but when used together we can generally use the singular.